### PR TITLE
Move resume operations out of ThreadFront (FE-1476)

### PR DESCRIPTION
**Pull request by hbenl**

### CODE REVIEW

File: src/ui/utils/resumeOperations.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    sourceId,
+    locationsToSkip,
+  }: ResumeOperationParams & { point: ExecutionPoint }
+) {
+  if (
+    loadedRegions.loaded.every(
+      region =>
+        BigInt(point) < BigInt(region.begin.point) || BigInt(point) > BigInt(region.end.point)
+    )
+  ) {
```
Comment (bvaughn) [inline src/ui/utils/resumeOperations.ts:43]: nit: We have utility methods for this:

```suggestion
  if (!isRangeInRegions(region.begin, loadedRegions.loaded)) {
```

File: src/devtools/client/debugger/src/reducers/pause.ts [hunk 2]
```diff
@@ -129,7 +130,7 @@ export const executeCommandOperation = createAsyncThunk<
   const symbols = sourceId ? await sourceOutlineCache.readAsync(replayClient, sourceId) : undefined;
   const nextPoint = await getResumePoint(replayClient, state, command);
 
-  const resp = await ThreadFront[command]({
+  const resp = await resumeOperations[command](replayClient, {
```
Comment (bvaughn) [inline src/devtools/client/debugger/src/reducers/pause.ts:133]: Pleasantly surprised to see that TypeScript covers this without falling back to an `any` type  :)

**Review by bvaughn — APPROVED** [review]

File: src/ui/utils/resumeOperations.ts (hunk 1 above)
Comment (bvaughn) [reply src/ui/utils/resumeOperations.ts:43]: Seems worth making this change?

File: src/ui/utils/resumeOperations.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    return null;
+  }
+
+  await ThreadFront.ensureAllSources();
```
Comment (markerikson) [inline src/ui/utils/resumeOperations.ts:47]: :hand: Can we kill this too? Surely we can use our sources cache or something instead.

Comment (hbenl) [reply src/ui/utils/resumeOperations.ts:47]: Yeah, that'll be part of my next PR where I clean up how we handle sources.

**Pull request merged**